# Chapter 4: Relationship between climate and fitness of a highly invasive succulent 

Authors:

Sam C. Levin 

Roberto Salguero-Gómez 

Tiffany M. Knight 

### Abstract

Invasive species pose a substantial threat to biodiversity, and Mediterranean dunes are especially threatened by them. In conjunction with biological invasions, climate change imposes further pressures on these ecosystems. In order to mitigate these threats, we need detailed information on how climate influences the fitness of invaders so that management efforts can be targeted to prevent future invasions and curb rapidly growing populations that are causing environmental harm. _Carpobrotus_ species are a widely distributed invasive species, making them ideal for studying how climate affects fitness. We developed a novel, drone based method of collecting demographic data for 13 sites distributed across the native and invaded ranges, and used this data to parameterize Integral Projection Models to estimate fitness as a function of climate variables. A life table response experiment on these models indicates that climate is not a strong driver of fitness.

```{r echo = FALSE, message = FALSE, warning = FALSE, eval = TRUE}

# Remove this block to compile complete dissertation.
library(dplyr)
library(tidyr)
library(tools)
library(kableExtra)
library(brms)
library(ggplot2)
library(randomForest)
library(patchwork)
library(forcats)
library(sf)

```


```{r echo = FALSE, warning = FALSE, message = FALSE}

rams <- readRDS("../Data/all_ramets_di.rds")
t_2_out <- read.csv("../Data/t_2_omissions.csv",
                    stringsAsFactors = FALSE) %>%
  setNames(c("site", "id"))

rams_t_2 <- anti_join(rams, t_2_out, by = c("site", "id"))

grow_mod <- readRDS("../Data/ramet_growth_list_krig.rds")[[1]][["times_sw3_seas"]]
grow_dat <- grow_mod$data

surv_mod <- readRDS("../Data/ramet_survival_list_krig.rds")[[1]][["times_sw3_seas"]]
surv_dat <- surv_mod$data

repr_mod <- readRDS("../Data/ramet_repro_list_krig.rds")[[1]][["times_sw1_seas"]]
repr_dat <- repr_mod$data

flow_mod <- readRDS("../Data/ramet_flower_n_list_lin_slope_ran.rds")[[1]][["times_sw3_seas"]]
flow_dat <- flow_mod$data

rf <- readRDS("../Data/all_site_rf_ltre_with_site_term.rds")
clim <- read.csv("../Data/site_climate_values.csv")
lams <- read.csv("../Data/lin_mod_general_ipm_site_lambdas_prob_resamp.csv")


```

### Introduction

Invasive species are ecological pests at local, regional, and global scales (Pimentel et al. 2005, Vila et al. 2011, Levin et al. 2020). There is an extensive body of work devoted to understanding how species become invasive (reviewed in Lowry et al. 2013 and Jeschke & Heger 2018). However, generalization across families, genera, and even species remains challenging as environmental contexts can vary greatly across the invaded range, and population level processes (e.g. founder effects, rapid evolution, or hybridization) can induce rapid change in the invading population (reviewed in Prentis _et al._ 2008). The environmental context of invasive species is receiving increasing attention, as there is growing recognition that climate change may favor invasive populations and produce an even greater threat to ecosystems (Beaury et al. 2020, ).

There is a growing body of literature illustrating the importance of the interaction between climate change and invasions. However, there are contrasting expectations for species that depend both on biome and the species in question. For example, Bradley and colleagues (2009) predicted that _Centauria solstitialis_ and various _Tamarix spp_ should undergo range expansion, while _Bromus tectorum_ should undergo range contraction. Similarly, Merow & colleagues (2017) found that _Allaria petiolata_ will likely experience range contraction, but _Berberis thumbergii_ will likely undergo range expansion under future climate conditions. Furthermore, using a species distribution modeling approach, Allen & Bradley (2016) found that invasive plant species richness would decline by ~7% by 2050. Despite the effort dedicated to studying this, there is no clear consensus on when climate change should cause range expansion or contraction. Furthermore, there are substantial gaps left in our knowledge of how the many facets of climate change (e.g. increased temperature, variance in precipitation, $CO_2$ enrichment) will affect various species and biomes (reviewed in Ziska 2022).

Given the joint issues posed by climate change and invasions, there have been recent efforts aimed at understanding their interactions. Species distribution models (SDMs) are often applied to this problem, as they make use of species occurrence and environmental covariate data to estimate species' current and future ranges (Guisan & Thullier 2005, Bradley et al. 2009, Allen & Bradley 2020). However, SDMs have also been criticized for their application applications, as they are often unable to predict occurrences in environments that are not covered by the observed range of the species in question (Merow et al. 2017, reviewed in Liu et al. 2020 and in Lake et al. 2020). Classical SDMs make use of presence data or presence-absence data for a species, and do not necessarily include information on spatial variation in population performance. Furthermore, making predictions on invaded range occupancy and performance based solely on data from the native range is risky because of novel biotic interactions that may arise, and because this practice often involves extrapolating estimates beyond the observed range of environmental data (Broennimann & Guisan 2008). Finally, the assumption that species are at equilibrium with their environment and thus occupy all suitable areas is unlikely to be true when using data from the native range (Elith et al. 2010).

One approach to handling these issues is to incorporate demographic data from both the native and invaded range, and sample broadly across environmental gradients (Merow et al. 2017, Briscoe et al. 2019). This approach explicitly links vital rates (e.g. survival, reproduction, and dispersal) to environmental conditions, allowing for a mechanistic approach to making performance projections (Briscoe et al. 2019). A drawback of this method is that it entails sampling populations on multiple continents and combining data from multiple sources, which is laborious and expensive. However, new methods of rapidly collecting demographic data, such as unmanned aerial vehicles (UAVs, drones) can reduce those burdens.Indeed, a number of authors have already used this technology to generate estimates of map populations of invasive species and estimate their fitness (Innangi et al. 2023, Bogdan et al. 2020, reviewed in Cavendar-Bares et al. 2022).

Mediterranean biomes are threatened by both climate change and invasions (Klausmeyer & Shaw 2009). These biomes are home to vast array of endemic plant and animal species. The Mediterranean ecosystems are threatened by the _Carpobrotus_ genus (Aizoaceae), particularly _C. edulis_, _C. acinaciformis_, _C. chilensis_, and their hybrids, which are native to South Africa but have been introduced on six continents. These _Carpobrotus_ species are spreading, mat forming succulents that are exceptionally competitive for water and light (D'Antonio & Mahall 1991, Novoa et al. 2012, Campoy et al. 2018). There is a growing body of evidence indicating these species are allelopathic, with their dessicated litter acidifying the soil and preventing other species from germinating (Novoa et al. 2012, Novoa et al. 2014).

Here, we apply novel demographic sampling methods to the _Carpobrotus_ genus across `r length(unique(rams$site))` sites spread throughout their native and invaded range. We use drones as a fast and cheap way of generating detailed maps of each population. We repeat sampling of populations across two years to estimate vital rates as a function of climate variation. Mediterranean ecosystems can experience strong water limitation, and _carpobrotus_ is exceptionally competitive for water resources (D'Antonio & Mahall 1991, Novoa et al. 2012, Campoy et al. 2018). We expect that this combination of factors will lead to reduced competition at water limited sites, because _Carpobrotus_ will exclude inferior competitors for this essential nutrient. Therefore, we expect that _Carpobrotus_ populations will have higher per-capita growth rates in environments that receive less precipitation. We use a combination of Integral Projection Models (IPMs), generalized additive models (GAMs), and a life table response experiment (LTRE) to test these hypotheses.

### Methods

This study was conducted across `r length(unique(rams$site))` sites spread across South Africa (native range), New Zealand (invaded), Portugal (invaded), and Israel (invaded, Figure 4.1). Sites in the Northern Hemisphere were sampled in March/April, and sites in the Southern Hemisphere were sampled in September/October each year so that plants were observed during their peak flowering period (i.e. early spring). The primary goal of this study was to understand how environmental variation affects _Carpobrotus_ population dynamics. Therefore, we chose geographic areas to sample based on maximizing the amount of climate space that we would sample with our study. Within each geographic area, we chose sites based on the extent of the population (i.e. had enough ramets to perform statistical analysis on), and the ability to sample without disturbing local flora and fauna (i.e. was not home to threatened/endangered bird species, Table 4.1). 

We used drones to collect aerial imagery each population. The following methods are general, and further site specific modifications to this protocol are described in Table 1 and Appendix 7. We developed flight plans to map each population using DJI Ground Station Pro v2 (henceforth called "GSP", SZ DJI Technology Co.) for iPad (Apple Inc). We used the app to draw polygons over an area of interest, and then let the app compute the optimal flight path given our desired resolution. We selected a flight path that generated a resolution of 0.2 - 0.45 cm/pixel. This resulted in a range of 70-85% side  and front photo overlap. The average width of open flowers is 7.46 cm (+/- 0.008 S.E.) and unripe fruits was 2.32 cm (+/- 0.002 S.E.) across all populations (SC Levin, unpublished data), so this resolution is sufficient to allow us to mark the majority of both in the resulting maps (orthomosaics, see below for more details). Transects were flown with a DJI Phantom 4 Pro v1 (SZ DJI Technology Co.) along the lines of the flight plan generated by DJI GSP and images were recorded at 1.8 - 2m intervals. When batteries reached a critical level of power (i.e. ~20%), we landed the drone, switched the batteries out, and the flight was resumed from the last stopping point. 

Once all images were captured after the first aerial survey, individual photos were processed into a single composite, georeferenced orthomosaic map using Pix4Dmapper (Pix4D SA, 2019). We drew individual polygons around contiguous *Carpobrotus* ramets, assigned each a unique ID number, and counted flowers using a point layer in QGIS (QGIS Development Team 2021). We repeated the process in 2019 by overlaying the 2019 map on the 2018 map and marking polygons for all surviving individuals as well as new recruits also in QGIS. The surveys sometimes did not perfectly overlap with each other, so ramets from the first survey that were not flown over again were excluded from all analyses. Individual ramets that were within the range of space sampled in both time periods but not found again in our second aerial survey were assumed dead. Maps were aligned between the first and second surveys using the Georeferencer GDAL plugin (QGIS Development Team 2021) in QGIS because the georeferencing of the orthomosaics was only accurate to ~5m. This plugin allows users to manually find and mark common points on both images and set them as reference points for transformation between the old coordinate system (map coordinates of survey 2) and the new coordinate system (map coordinates of survey 1). We used the thin plate splines transformation for the coordinate systems and nearest neighbor resampling method. The reference points are available in the supplementary materials. In addition to drawing polygons around the individual plants, we drew polygons around targets of known sizes and computed the ratio of calculated areas under the polygon vs the known sizes. We then re-scaled all computed sizes of plants using this ratio. Once all sizes, flowering information, and survival data were digitized, we used the `sf` (Pebesma 2018) and `dplyr` (Wickham et al. 2022) _R_ packages to join all the ramet data into a single dataset. In some cases, the second drone flight could not cover the entire extent of the first one (e.g. due to new construction of power lines). Ramets that could not be re-sampled were included in the probability of flowering and number of flowers produced regression models, but omitted from survival and growth analyses (see below for details).

```{r echo = FALSE, message = FALSE, warning = FALSE}

ipm_sites <- c("Struisbaai", "St_Francis", "Melkboss", "Vogelgat",
               "Rooisand", "Springfontein", "Rough_Island",
               "Rarangi", "Foxton", "Whirinaki", "Havatselet",
               "Praia_de_Areao", "Colares")

sites <- read.csv("../Data/drone_flight_metadata.csv",
                  stringsAsFactors = FALSE) %>%
  filter(population %in% ipm_sites)


sites <- rams %>%
  group_by(site) %>%
  summarise(N_ramets = n()) %>%
  right_join(sites, by = c("site" = "population")) %>%
  select(site, lat, lon, date_flight_1, date_flight_2, N_ramets) %>%
  mutate(lat = round(lat, 2), lon = round(lon, 2))

sites$site <- gsub("_", " ", sites$site)

names(sites) <- gsub("_", " ", names(sites))

names(sites) <- toTitleCase(names(sites))

cap <- "Table 4.1: Population locations, drone flight dates, and the number of unique ramets identified in orthomosaic generated from the first flight. "

kable(sites,
       format = "simple",
      caption = cap)# %>%
  # column_spec(1, border_left = TRUE) %>%
  # column_spec(6, border_right = TRUE)

```

```{r echo = FALSE, message = FALSE, warning = FALSE, fig.height = 8, fig.width = 10, fig.cap = "**Figure 4.1**:  Geographic distribution of all sites used for climate envelope calculations (in black) and all sites used for demographic modeling (in red). Insets show locations within each region."}


wrld <- st_read("../Data/World_Continents/continents.shp", quiet = TRUE) 

all_clim_sites <- read.csv("../Data/all_gbif_field_sites.csv",
                           stringsAsFactors = FALSE)  %>%
  filter(site %in% ipm_sites | is.na(site)) %>%
  mutate(sampled_for_demography = case_when(
    site %in% ipm_sites ~ "Yes",
    TRUE ~ "No"
  ))

sa <- filter(all_clim_sites, site %in% ipm_sites[1:6])
nz <- filter(all_clim_sites, site %in% ipm_sites[7:10])
med <- filter(all_clim_sites, site %in% ipm_sites[11:13])

med_map <- ggplot(wrld,
                  aes(group = CONTINENT)) +
  geom_sf(color = "grey50",
          fill = "antiquewhite") +
  coord_sf(xlim = c(-12, 36),
           ylim = c(30, 42)) +
  geom_point(data = med, 
             aes(x = lon, y = lat), 
             color = "red",
             inherit.aes = FALSE) +
  theme(panel.background = element_rect(fill = "aliceblue"),
        axis.title = element_blank(),
        plot.margin = unit(c(0, 0, 0, 0), "mm"),
        axis.text = element_blank(),
        axis.ticks = element_blank(),
        panel.grid = element_blank(),
        axis.ticks.length = unit(0, "pt")) +
  annotate("text", x = 2, y = 32, size = 4, label = "EU/Israel")

sa_map <- ggplot(wrld,
                  aes(group = CONTINENT)) +
  geom_sf(color = "grey50",
          fill = "antiquewhite") +
  coord_sf(xlim = c(15, 29),
           ylim = c(-38, -30)) +
  geom_point(data = sa, 
             aes(x = lon, y = lat), 
             color = "red",
             inherit.aes = FALSE) +
  theme(panel.background = element_rect(fill = "aliceblue"),
        axis.title = element_blank(),
        plot.margin = unit(c(0, 0, 0, 0), "mm"),
        axis.text = element_blank(),
        axis.ticks = element_blank(),
        panel.grid = element_blank(),
        axis.ticks.length = unit(0, "pt")) +
  annotate("text", x = 22, y = -35.5, size = 4, label = "South Africa")

nz_map <- ggplot(wrld,
                  aes(group = CONTINENT)) +
  geom_sf(color = "grey50",
          fill = "antiquewhite") +
  coord_sf(xlim = c(165, 180),
           ylim = c(-48, -36)) +
  geom_point(data = nz, 
             aes(x = lon, y = lat), 
             color = "red",
             inherit.aes = FALSE) +
  theme(panel.background = element_rect(fill = "aliceblue"),
        axis.title = element_blank(),
        plot.margin = unit(c(0, 0, 0, 0), "mm"),
        axis.text = element_blank(),
        axis.ticks = element_blank(),
        panel.grid = element_blank(),
        axis.ticks.length = unit(0, "pt")) +
  annotate("text", x = 172, y = -43, size = 3.5, label = "New Zealand")

world_map <- ggplot(wrld,
                   aes(group = CONTINENT)) +
  geom_sf(color = "grey50",
          fill = "antiquewhite") + 
  coord_sf(xlim = c(-180, 180),
           ylim = c(-60, 60), expand = FALSE) +
  geom_point(data = all_clim_sites, 
             aes(x = lon, y = lat,
                 color = sampled_for_demography,
                 alpha = sampled_for_demography,
                 size  = sampled_for_demography), 
             inherit.aes = FALSE) +
  xlab("") +
  ylab("") +
  scale_color_manual(values = c("red", "black"),
                     breaks = c("Yes", "No")) +
  scale_size_manual(values = c(2, 1.25),
                    breaks = c("Yes", "No")) +
  scale_alpha_manual(values = c(0.7, 0.3),
                     breaks = c("Yes", "No")) +
  guides(size  = guide_legend("Sampled For Demography"),
         color = guide_legend("Sampled For Demography"),
         alpha = guide_legend("Sampled For Demography")) +
  theme(legend.position = "top",
        legend.direction = "horizontal",
        panel.background = element_rect(fill = "aliceblue"),
        plot.margin = unit(c(0, 0, 0, 0), "mm"),
        axis.text = element_blank(),
        axis.ticks = element_blank()) 

world_map + 
  inset_element(med_map, 
                left = 0,
                top = 0, 
                bottom = -0.3, 
                right = 0.25)  + 
  inset_element(sa_map, 
                left = 0.3,
                top = 0, 
                bottom = -0.35, 
                right = 0.7) +
  inset_element(nz_map, 
                left = 0.75,
                top = 0,
                bottom = -0.4, 
                right = 1)

```

The primary goal of this study was to understand how environmental variation constrains or facilitates invasion by _Carpobrotus_ species. Therefore, we compiled a data set of georeferenced _Carpobrotus_ records from across the globe from GBIF, and cleaned them with the `CoordinateCleaner` _R_ package (Zizka et al. 2019, Figure 4.1). We downloaded monthly ERA5 climate data from 2018 - 2020 for temperature ($\theta_t$), precipitation ($\theta_p$), and soil water content ($\theta_{s,i}$ where $i$ is the soil layer with values 1-3 to denote 0-7cm, 7-30cm, and 30-100cm deep, respectively). Data were kriged for each occurrence point, plus our site coordinates using the `krigR` _R_ package (Kusch & Davy, 2022). We aggregated data by wet and dry season (i.e. $\theta_s$ values for the wet season, and $\theta_s$ values for the dry season). Finally, we computed how many standard deviations each of our sites was from the global mean value of each environmental covariate to create a standardized environmental value for each site in our data set (Compagnoni et al. 2021).  

We modeled size-dependent survival (Bernoulli model with logit link, $s_a(z, \theta)$), size  at time $t + 1$ ($z'$) conditional on survival and size at time $t$ (Gaussian model with identity link and size dependent variance, $G(z'|z, \mu_g, \sigma_g, \theta)$), size-dependent probability of flowering (Bernoulli model with logit link, $p_f(z, \theta)$), and size-dependent flower production (negative binomial model with log link, $r_f(z, \theta)$). In addition to size, we included environmental covariates ($\theta$, described in previous paragraph). We constructed Bayesian generalized linear mixed models (GLMMs) and compared different structures using WAIC, posterior predictive checks, and visual inspection of predictions against the observed data. We selected the best models chosen by WAIC that also made realistic predictions as determined by the latter graphical checks. Each of the potential forms used $\theta_t, \theta_p$ and one of the $\theta_{s,i}$ values as predictors, and interactions between size and environmental covariates were tested as well (included in the GAMMs as tensor product interactions). Because the responses of $p_f(z, \theta)$ and $r_f(z, \theta)$ are based on state at time $t$, we included environmental covariates from the year before the initial demographic sampling (i.e. $t-1$ to $t$), whereas $s_a(z, \theta)$ and $G(z'|z, \sigma, \theta)$ to $t+1$ used environmental data from the year between demographic samples. All candidate models included a population-specific intercept term and interaction between size and nativity. All vital rate models were implemented with the `brms` _R_ package (Bürkner et al. 2018).

In addition to the vital rates above, we also modeled the size of new recruits at time $t + 1$ ($r_a(z')$), and estimated probabilities for flower pod survival ($p_{fv}$) and maturation ($p_m$) to time $t+1$, immediate seed germination ($g_i$), total viability ($v_s$), and seed establishment probability ($p_e$), and seed bank germination ($g_{sb}$) and survival ($s_{sb}$).  Data for $r_a(z')$ was collected from the Rooisand site in South Africa. In 2018, we delineated 2 $40 cm \times 40 cm$ plots and counted the number of seedlings. In 2019, we returned and flew brief surveys with the drone at ~ 1.2m above ground level. We generated orthomosaics using the same procedure described above, and then counted surviving seedlings and drew polygons around them to estimate their size. Unfortunately, we did not find seedlings in sufficient quantities at other sites to establish plots, and so our sample size is limited to these two plots at a single site for these parameters. We conducted literature reviews to gather estimates of other seed-related parameters, and these were estimated by averaging over the reported estimates (Table 4.2). For two vital rates ($p_e$ and $s_{sb}$), we could not find values in the literature. Therefore, we conducted simulations to examine a reasonable range of values ($1^{-5} - 1^{-4}$ and $1^{-2} - 1^{-1}$, respectively) to examine how our results changed as a function of these values. Our conclusions are not sensitive to these values, and these results are reported in Appendix 7.

We used the parameters and functions described above to generate an Integral Projection Model (IPM) for each site (Ellner, Childs & Rees 2016). Our IPM included 1 continuous trait, size ($z, z'$), and 3 discrete states - mature fruits ($mf$), seedlings ($sdl$), and seeds in the seed bank ($sb$) (see Appendix 7 for complete equations and parameter distributions describing the IPM). We combined the vital rate functions described above into an IPM in _R_ using the `ipmr` package (Levin et al. 2021). Point estimates of the deterministic per-capita growth rate for each site, $\lambda_i$ were computed from IPMs parameterized with mean values of the posterior distribution for each parameter estimated in the Bayesian regression models described above. We then used the posterior distributions for each regression model to create 4000 $\lambda_i$ values representing a range of plausible IPMs for each site. 

Our goal was to determine both the shape and magnitude of a climate driver's effect on $\lambda$ _and_  the relative importance of each driver for _Carpobrotus_'s fitness at a global scale. To address the first goal, we constructed GAMs with the estimated $\lambda$ values as the response and each climate driver as a predictor. GAMs are well suited to this because population responses to a climate driver may be non-linear across the range of sampling. To address the second goal, we conducted an Life Table Response Experiment (LTRE) with $\lambda$ as the response and the spatially varying IPM parameters as predictors (Caswell 2001, Ellner, Childs & Rees 2016). Specifically, we used a random forest to estimate the relationship between the posterior draws for $\lambda$ and the IPM regression parameters. Random forests are well suited for this analysis because they are non-parametric, non-linear models that can handle high dimensional data (Breiman 2001). We assessed the importance of each variable by randomizing parameter values and computing the change in mean squared error (MSE) of the of the overall regression tree. Larger increases in MSE for a given parameter value indicate that it is more important in predicting $\lambda$, and therefore contributing more to the observed value of $\lambda$ (Ellner, Childs & Rees 2016). We implemented all of the LTRE analyses using the `randomForest` _R_ package (Liaw & Weiner 2002).

### Results

We successfully sampled `r nrow(repr_dat)` ramets for probability of flowering ($p_f(z, \theta)$), `r nrow(flow_dat)` ramets for number of flowers produced ($r_f(z, \theta)$), `r nrow(surv_dat)` ramets for survival ($s_a(z, \theta)$), and `r nrow(grow_dat)` ramets for growth ($G(z' | z, \sigma, \theta)$). Model selection and diagnostic plots for all vital rates indicated GLMM fits were best. 

IPMs yielded $\lambda$ values between 0.6 and 1.3 for the 13 populations. GAMs for $\lambda_i$ as a function of each climate variable show differing responses, though no relationship was statistically significant. Climate parameters were generally of low importance in predicting $\lambda$ values, yielding little evidence of a relationship between climate and population performance (Figure 4.2). Dry season precipitation showed the most negative, yet non-significant, effect on $\lambda_i$ ($p = 0.243$, soil water content results not shown here). All other regressions of $\lambda$ on a climate driver were not significant.

```{r, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = "**Figure 4.2**: Relationship between $\\lambda$ and the climate variables included in the IPMs. Black dashed lines represent the fit from the GAMs of `lambda ~ <climate_variable>`, and red dotted lines indicate $\\lambda = 1$, meaning the population is stable. None of the relationships shown here are statistically significant.", fig.height = 8, fig.width = 10}

all_lams <- lams %>%
  pivot_longer(cols = -c(obs),
               names_to = "site",
               values_to = "lambda") %>%
  mutate(site = gsub("lambda_", "", site))

clim_vars <- expand.grid(c("temp", "prec", "sw1", "sw3"), c("dry", "wet"), c("t", "t_1")) %>%
  apply(1, function(x) paste(x, collapse = "_")) %>% 
  c("site") %>%
  syms()


all_clim <- clim %>%
  select(!!! clim_vars) %>%
  pivot_longer(-site,
               names_to = "clim_var", 
               values_to = "clim_val")

all_data <- all_lams %>%
  group_by(site) %>%
  summarise(obs_val  = first(lambda[obs == "yes"]),
            up_ci    = quantile(lambda[obs == "no"], 0.95),
            lo_ci    = quantile(lambda[obs == "no"], 0.05)) %>%
  left_join(all_clim) %>%
  mutate(native = case_when(
    site %in% c("Struisbaai", "Rooisand",
                "Vogelgat", "Springfontein",
                "Melkboss", "St_Francis") ~ "Native",
    TRUE ~ "Invasive"
  ))

use_clim_vars <- expand.grid(clim = c("prec_dry", "prec_wet"),
                             t    = c("t", "t_1")) %>%
  apply(1, FUN = function(x) paste0(x, collapse = "_"))

plt_dat <- filter(all_data, clim_var %in% use_clim_vars) %>%
  mutate(clim_var = case_when(
    clim_var == "prec_dry_t"   ~ "Dry season precipitation (t)",
    clim_var == "prec_wet_t"   ~ "Wet season precipitation (t)",
    clim_var == "prec_dry_t_1" ~ "Dry season precipitation (t - 1)",
    clim_var == "prec_wet_t_1" ~ "Wet season precipitation (t - 1)"
  ), 
  clim_var = fct_relevel(as.factor(clim_var),
                         c("Dry season precipitation (t)",
                           "Wet season precipitation (t)",
                           "Dry season precipitation (t - 1)",
                           "Wet season precipitation (t - 1)"
                           )))

ggplot(plt_dat,
              aes(x = clim_val, y = obs_val, color = site, shape = native)) +
  geom_hline(yintercept = 1, color = "firebrick", linetype = "dotted", 
             size = 1.2) + 
  geom_point(size = 5, alpha = 0.5) +
  geom_linerange(aes(ymin = lo_ci, ymax = up_ci), size = 1.2, alpha = 0.5) +
  geom_smooth(aes(x = clim_val, y = obs_val),
              inherit.aes = FALSE,
              method = "gam",
              formula = y ~ s(x, k = 5),
              color = "black",
              linetype = "dashed") +
  theme_bw() +
  facet_wrap(~ clim_var, scales = "free_x") +
  xlab("Climate Value (SD from global mean)") +
  ylab(expression(lambda))


```

The random forest LTRE explained `r round(rf$rsq[751], 2) * 100`% of the variance in $\lambda$ across sites. Decomposing the contributions of each parameter showed that site specific random effects were the most important parameters in the model. Climate related vital rate parameters were of substantially lesser importance, but many were included in the top 20 most important parameters in the IPMs (Figure 4.3).


```{r echo = FALSE, message = FALSE, warning = FALSE, fig.cap = "**Figure 4.3**: Variable importance as determined by the random forest. Climate-related parameters are orders of magnitude less important than site-specific random effects, indicating that unmeasured variation is explaining more variance in $\\lambda$.", fig.height = 8, fig.width = 10}


imps <- rf$importance %>%
  as.data.frame() %>%
  mutate(var = rownames(.)) %>%
  select(c(1, 3))%>% 
  cbind(imp_sd = rf$importanceSD) %>%
  setNames(c("inc_mse", "var", "imp_sd")) %>%
  filter(!grepl("sw[0-9]", var)) %>%
  arrange(desc(inc_mse)) %>%
  mutate(var = gsub("flow", "Flower #", var),
         var = gsub("repr", "Pr(flowering)", var),
         var = gsub("grow", "Growth", var),
         var = gsub("surv", "Survival", var),
         var = gsub("slope_ran", "Random Slope", var),
         var = gsub("intercept_ran", "Random Intercept", var),
         var = gsub("temp", "Temperature", var),
         var = gsub("x", "X", var),
         var = gsub("int", "Fixed Intercept", var),
         var = gsub("prec", "Precipitation", var),
         var = gsub("log_size", "Size", var),
         var = gsub("__", "_", var),
         var = gsub("_", " ", var),
         var = gsub("wet", "Wet", var),
         var = gsub("dry", "Dry", var),
         var = gsub("sigma", "Sigma", var),
         var = gsub("native", "Nativity", var)) %>%
  mutate(var = factor(var, levels = as.character(var)),
         var = fct_reorder(var, inc_mse, .fun = max))


ggplot(filter(imps, var != "site")[1:20, ], 
                            aes(x = var, y = inc_mse)) + 
  geom_point(size = 2) + 
  geom_linerange(aes(ymin = pmax(0, inc_mse - 2 * imp_sd), ymax = inc_mse + 2 * imp_sd),
                 size = 1.2) +
  coord_flip() +
  theme_bw() +
  ylab("Importance") +
  xlab("Vital Rate Regression Parameter")



```

### Discussion

Sites varied considerably in their population growth rates (Figure 4.2). However, much of the variation $\lambda$ is explained by unmeasured variables - the LTRE results showed that the random intercepts and slopes contribute the most to explaining variance in $\lambda$ (Figure 4.3). While climate factors were much less important than random variables in the LTRE, sites with lower dry season precipitation appeard to have higher $\lambda$ values (Figure 4.2). This result is not statistically significant and does not yield robust support for our hypothesis that _Carpobrotus_ has higher performance in drier sites.

Many factors may contribute to a species success at local scales and might explain why unmeasured variables contribute more to the variation in $\lambda$. First, phylogenetic and functional distinctiveness of _Carpobrotus_ to the local native plant community may have a greater fitness impact than abiotic conditions when species are introduced to novel settings. Distinctiveness has been shown to allow species to have novel resource use and compete less strongly with native species, thus decreasing the effects of competitors on vital rates and $\lambda$ (Levin et al. 2020). Second, _Carpobrotus_ was intentionally introduced in much of its invaded range, largely for ornamental purposes (Portela et al. 2022). Horticultural introductions usually entail human preference for desirable characteristics, such as heavy flower production, which are also desirable characteristics for an invader that escapes cultivation (Ca$\mathrm{\tilde{n}}$o et al. 2008). Such anthropogenic selection may induce founder effects for more fit phenotypes than are present in its native range, allowing it to become more dominant (Mayr 1942). Third, we note that the introduced range sometimes contains multiple _Carpobrotus_ species and their hybrids (Suehs et al. 2004). The high fitness in a few of the populations could be the result of hybridization between multiple introduced populations could yield super-hybrids (Novoa et al. _in prep_). 

The successional stage of the _Carpobrotus_ population might also contribute to the unexplained site variation in $\lambda$. _Carpobrotus_ tends to be an early successional species in its native range, losing out to other Fynbos species as the ecosystem matures after fire (T. Masayiti, personal communication). However, the dense litter that forms under plants as they grow inhibits germination of native species and alters the soil geochemistry in the invaded range, which may delay or halt succession in habitats where native species are not adapted to the presence of _Carpobrotus_ (Santoro *et al.* 2011, Novoa *et al.* 2013, Compoy *et al.* 2018). An alteration of the successional trajectory of the ecosystem and the associated competitive release that comes with that could easily mask the smaller effect of climate on overall fitness. Documenting the succesional stage of populations cannot be easily done post-hoc the orthomosaic outputs from the drone do not preserve the vertical structure of the populations.

In contrast to expectations, fitness was not significantly higher in the introduced range than the native range. Both the best and worst performing populations were present in the introduced range. For example, Praia de Areao was the highest performing, and is situated in full sun along the Portuegese coast. This is a relatively mild winter climate, which may be helpful to the population. In contrast, Foxton is located on the southwestern side of New Zealand's North Island. Strong winds, cooler temperatures, and shifting dunes are all factors that may contribute to this site's relative lack of performance.

Future work investigating climate drivers of invasions should consider that spatial environmental gradients alone are not sufficient for robust statistical inference. We specifically recommend replicating sites both within _and_ across climate bands. A recent simulation study illustrated that replicating 10 sites within a climate band for 3 years provides as much power for detecting climate-demography relationships as a 20-30 year data set at a single site (Compagnoni et al. _unpublished_). Our study design only yielded three sites that could serve as climate replicates of each other, while all other sites were spread across a considerable distances and were in different climate bands. This yielded a broad climate gradient, but also did not allow partitioning of site specific variability and climate signal (because $n = 1$ for every level of climate in our data set). 

Concerns that climate change may favor invasive species over native ones are not supported by this study. Climate is not found to be a strong predictor of demographic success for _Carpobrotus_. This could be because anthropogenic factors are stronger predictors of invasion success than ecological ones (Wohlwend et al. 2021). While climate change may open up new areas to invasion, long distance dispersal events are still required for a species to realize this new niche space. In the Anthropocene era, these dispersal events are far more likely to occur due to economic factors than demographic ones, as evidenced by the fact that initial introductions of _Carpobrotus_ were almost exclusively intentional throughout the invaded range. It is therefore more important to focus on limiting trade of these species to curtail spread at the macroecological scale.

### Citations

1. Bürkner, P.-C. (2018). Advanced Bayesian Multilevel Modeling with the R Package brms. The R Journal, 10(1), 395-411. doi:10.32614/RJ-2018-017

2. Liaw, A. & Wiener, M. (2002). Classification and Regression by randomForest. R News 2(3), 18--22.

3. Breiman, L. (2001). Random Forests. Machine Learning 45: 5-32. DOI: https://doi.org/10.1023/A:1010933404324

4. Pebesma, E., 2018. Simple Features for R: Standardized Support for Spatial Vector Data. The R Journal 10 (1), 439-446, https://doi.org/10.32614/RJ-2018-009

5. Wickham, H., François, R.,  Henry, L., & Müller, K. (2021). dplyr: A Grammar of Data
  Manipulation. R package version 1.0.7. https://CRAN.R-project.org/package=dplyr

6. Zizka, A., Silvestro, D., Andermann, T., Azevedo, J., Duarte Ritter, C., Edler, D., _et al._ (2019). “CoordinateCleaner: standardized cleaning of occurrence records from biological collection databases.” Methods in Ecology and Evolution 10. DOI: 10.1111/2041-210X.13152 (URL: https://doi.org/10.1111/2041-210X.13152), R package version 2.0-20

7. Portela, R., Barreiro, R., Alpert, P., Xu, C-Y., Webber, B.L., & Roiloa, S.R. (2022). Comparative invasion ecology of _Carpobrotus_ from four continents: responses to nutrients and competition. Jourcnal of Plant Ecology, _in press_. https://doi.org/10.1093/jpe/rtac034

8. Wohlwend, M.R., Craven, D., Weigelt, P., Seebens, H., Winter, M., Kreft, H., _et al_. (2021). Anthropogenic and environmental drivers shape diversity of naturalized plants across the Pacific. Diversity and Distributions. https://doi.org/10.1111/ddi.13260

9. Santoro, R., T. Jucker, M. Carranza, and A. T. R. Acosta. (2011). Assessing the Effects of Carpobrotus Invasion on Coastal Dune Soils. Does the Nature of the Invaded Habitat Matter? Community Ecology 12 (2): 234–240. doi:10.1016/j.jnc.2011.08.003.

10. Novoa, A., L. González, L. Moravcová, and P. Pyšek. (2013). Constraints to Native Plant Species Establishment in Coastal Dune Communities Invaded by Carpobrotus edulis: Implications for Restoration. Biological Conservation 164: 1–9. doi:10.1016/j.biocon.2013.04.008.

11. Campoy, J.G., Acosta, A.T.R., Affre, L., Barreiro, R., Brundu, G., Buisson, E. _et al._ (2018). Monographs of invasive plants in Europe: _Carpobrotus_. Botany Letters. https://doi.org/10.1080/23818107.2018.1487884

12. Pimentel D, Zuniga R, Morrison D. 2005 Update on the environmental and economic costs associated with alien-invasive species in the United States. Ecol. Econ. 52, 273–288. (doi:10.1016/j.ecolecon. 2004.10.002) 

13. Vila M et al. (2011). Ecological impacts of invasive alien plants: a meta-analysis of their effects on species, communities and ecosystems. Ecology Letters 14, 702–708. (doi:10.1111/j.1461-0248.2011.01628.x)

14. Levin, S.C., Crandall, R.M., Pokoski, T., Stein, C., & Knight, T.M. (2020). Functional and phylogenetic distinctiveness explain alien plant population responses to competition. Proceedings B 287: 20201070 http://dx.doi.org/10.1098/rspb.2020.1070

15. Lowry E, Rollinson EJ, Laybourn AJ, Scott TE, Aiello- Lammens ME, Gray SM, Mickley J, Gurevitch J. (2013). Biological invasions: a field synopsis, systematic review, and database of the literature. Ecol. Evol. 3, 182–196. (doi:10.1002/ece3.431)

16. Jeschke JM, Heger T. (2018). Invasion biology: hypotheses and evidence. Boston, MA: CABI Publishing.

17. Novoa, A., L. González, L. Moravcová, and P. Pyšek. (2012). Effects of Soil Characteristics, Allelopathy and Frugivory on Establishment of the Invasive Plant Carpobrotus edulis and a Co-Occuring Native, Malcolmia littorea. PLoS ONE 7 (12): e53166. doi:10.1371/journal.pone.0053166.

18. Novoa, A., and L. González. (2014). Impacts of Carpobrotus edulis (L.) NE Br. On the Germination, Establishment and Survival of Native Plants: A Clue for Assessing Its Competitive Strength. PLoS ONE 9 (9): e107557. doi:10.1371/journal.pone.0107557.

19. D'Antonio, C.M. & Mahall, B.E. (1991). Root Profiles and Competition between the Invasive, Exotic Perennial, Carpobrotus edulis, and Two Native Shrub Species in California Coastal Scrub. American Journal of Botany 78(7): 885-894.

20. Guisan, A. and Thuiller, W. (2005) Predicting Species Distribution: Offering More than Simple Habitat Models. Ecology Letters, 8, 993-1009. http://dx.doi.org/10.1111/j.1461-0248.2005.00792.x

21. Liu, C., Wolter, C., Xian, W., & Jeschke, J. (2020). Species distribution models have limited spatial transferability for invasive species. Ecology Letters 23(11): 1682-1692.

22. Merow, C., Bois, S.T., Allen, J.M., Xie, Y., Silander Jr., J.A. (2017). Climate change both facilitates and inhibits invasive plant ranges in New England. Proceedings of the National Academy of Sciences. E3276–E3284. www.pnas.org/cgi/doi/10.1073/pnas.1609633114 

23. Garcia-Callejas, D., Molowny-Horas, R., & Retana, J. (2017). Projecting the distribution and abundance of Mediterranean tree species under climate change: a demographic approach. Journal of Plant Ecology 10(5): 731-743.

24. Broennimann, O. & Guisan, A. (2008). Predicting current and future biological invasions: both native and invaded ranges matter. Biology Letters 4(5):585-589.

25. Elith, J., Kearney, M., & Phillips, S. (2010). The art of modelling range-shifting species. Methods in Ecology and Evolution 1(4): 330-342.

26. QGIS Development Team (2022). QGIS Geographic Information System. Open Source Geospatial Foundation Project. http://qgis.osgeo.org

27. Briscoe, N.J., Elith, J., Salguero-Gomez, R., Lahoz-Monfort, J.J., Carnac, J.S., Glijohann K.M. _et al._ (2019). Forecasting species range dynamics with process-explicit models: matching methods to applications. Ecology Letters 22: 1940-1956.

28. Ellner, S.P., Childs, D.Z., Rees, M. (2016) Data-driven modelling of structured populations: a practical guide to the integral projection model. Basel, Switzerland: Springer International Publishing AG

29. Levin, S.C., Childs, D.Z., Compagnoni, A., Evers, S., Knight, T.M., & Salguero-Gomez, R. (2021). ipmr: flexible implementation of Integral Projection Models. Methods in Ecology and Evolution 12 (10): 1826-1834.

30. Caswell, H. (2001) Matrix population models: construction, analysis, and interpretation, 2nd edn. Sunderland, MA: Sinauer Associates Inc

31. Prentis, P.J., Wilson, J.R.U., Dormontt, E.E., Richardson, D.M., & Lowe, A.J. (2008). Adaptive evolution in invasive species. Trends in Plant Science 13(6): 288-294.

32. Compagnoni A., Pardini, E., & Knight, T.M. (2021). Increasing temperature threatens an already endangered coastal dune plant. Ecosphere 12(3): e03454

33. Klausmeyer, K.R. & Shaw. M.R. (2009). Climate Change, Habitat Loss, Protected Areas and the Climate Adaptation Potential of Species in Mediterranean Ecosystems Worldwide. PLoS ONE 4(7): e6392. https://doi.org/10.1371/journal.pone.0006392

34. Ca$\mathrm{\tilde{n}}$o, L., Escarre, J., Fleck, I., Blanco-Moreno, J.M., & Sans, F.X. (2008). Increased fitness and plasticity of an invasive species in its introduced range: a study using *Senecio pterophorus*. Journal of Ecology 96: 468-476.

35. Beaury, E.M., Fusco, E.J., Jackson, M.R. et al. Incorporating climate change into invasive species management: insights from managers. Biological Invasions 22, 233–252 (2020). https://doi.org/10.1007/s10530-019-02087-6

36. Allen, J., & Bradley, B.A. (2016). Out of the weeds? Reduced plant invasion risk with climate change in the continental United States. Biological Conservation 203, 306-312. https://doi.org/10.1016/j.biocon.2016.09.015

37. Ziska, L.H. (2022) Plant Invasions, Rising CO2, and Global Climate Change. Chapter in Global Plant Invasions. 71-87. https://link.springer.com/chapter/10.1007/978-3-030-89684-3_4

38. Cavender-Bares, J., Schneider, F.D., Santos, M.J. et al. Integrating remote sensing with ecology and evolution to advance biodiversity conservation. Nature Ecology & Evolution. 6, 506–519 (2022). https://doi.org/10.1038/s41559-022-01702-5

39. Lake, T.A., Runquist, R.D.B, & Moeller, D.A. (2020). Predicting range expansion of invasive species: Pitfalls and best practices for obtaining biologically realistic projections. Diversity and Distributions 26(12): 1767-1779. 

40. Innangi, M., Marzialetti, F., Di Febbraro, M., Acosta, A.T.R., De Simone, W., Frate, L., Finizio, M., Perna, P.V., & Carranza, M.L. (2023). Coastal Dune Invaders: Integrative Mapping of Carpobrotus sp. pl. (Aizoaceae) Using UAVs. Remote Sensing. 15, 503. https://doi.org/10.3390/rs15020503
